main.js

```diff
@@ -122,29 +122,16 @@ function closePopup() {
   customLocationPopup.style.display = "none";
 
   if (selectedMainCity === 0) {
-    marker.setLatLng([-25.73134, 28.21837]);
-    map.setView([-25.73134, 28.21837], 8);
+    loadMarker(-25.73134, 28.21837);
   } else if (selectedMainCity === 1) {
-    marker.setLatLng([-26.195246, 28.034088]).addTo(map);
-    map.setView([-26.195246, 28.034088], 8);
+    loadMarker(-26.195246, 28.034088);
   } else if (selectedMainCity === 2) {
-    marker.setLatLng([-29.8579, 31.0292]).addTo(map);
-    map.setView([-29.8579, 31.0292], 8);
+    loadMarker(-29.8579, 31.0292);
   } else if (selectedMainCity === 3) {
-    marker.setLatLng([-33.918861, 18.4233]).addTo(map);
-    map.setView([-33.918861, 18.4233], 8);
+    loadMarker(-33.918861, 18.4233);
   }
 }
 
 // Add the click event to the close icon to trigger the close function.
 const closePopupButton = document.getElementById("close-popup-button");
```

Comment:
As above. Export the function elsewhere so you don't need to know _implementation_ details in your orchestrator.

Reply:
Moved to domManipulation.js.